source/slang/slang-parser.cpp

```diff
@@ -3871,6 +3872,20 @@ namespace Slang
         return parseBoolLitExpr(parser, false);
     }
 
+    static bool _isFinite(double value)
+    {
+        // Lets type pun double to uint64_t, so we can detect special double values
+        union
+        {
+            double d;
+            uint64_t i;
+        } u = { value };
```

Comment:
Is using `{}` initialization for a `union` well-defined? I've definitely seen it work, but had always assumed it was some compiler-specific language extension stuff.